k-distribution/src/main/scripts/bin/kparse-gen

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+set -u
+
+export PATH="$(cd "$(dirname "$0")"; pwd):$PATH"
+
+# initialize flags
+ARGV=()
+bisonFiles=()
```

Comment:
Initializing arrays as empty then appending to them later will fail on default Bash on MacOS unless we instruct users to update their Bash. It gives a weird error that we've had to debug for users several times.

Reply:
One trick I've used is to just initialize the array to something like `ARGV=(--)`, and then when using the array skip the first argument.

Reply:
As I said in the other PR, you are thinking of associative arrays; this is just a regular array and it works fine on Bash 3.